### ISSUE TITLE
Unable to set the property 'classList' on an element

### ISSUE THREAD

**Issue by jbdeboer** [angular/angular#876]
This needs to be implemented.
```
@Decorator({
  selector: '[class]',
  bind: {
    'classChanges': 'class | keyValueDiff'
  }
})
class ClassDecorator {
  element:NgElement;
  constructor(element:NgElement) {
    this.element = element;
  }

  classChanges(changes:KeyValueChanges) {
    changes.forEachChangedItem(...)
  }
}

@Decorator({
  selector: '[class-list]',
  bind: {
    'classChanges': 'classList | iterableDiff'
  }
})
class ClassDecorator {
  element:NgElement;
  constructor(element:NgElement) {
    this.element = element;
  }

  classChanges(changes:ArrayChanges) {
    changes.forEachChangedItem(...)
  }
}
```

### Original Text

I tried both

```html
<div [class-list]="'one two'"></div>
```
and

```html
<div [classList]="'one two'"></div>
```

at element_binder_builder.dart:158, the first example is checking the property 'class-list' on the JS element.
The second example is checking the property 'classlist'.

I would expect it to set the property 'classList'.

This is with the Dart release 2.0.0-alpha-12

**Comment by pkozlowski-opensource** [comment]
@jbdeboer you can't really write `[classList]="'one two'"` as a browser will see it as `[classlist]="'one two'"`  (lower-casing, HTML is not case sensitive).

Now, doing `<div [class-list]="'one two'"></div>` will be possible as soon as #866 is fixed. Having said this, binding to the `classList` property directly would mean that you overriding any CSS classes set before (for example, by the `class` attribute). This is why we are going to have [class] binding as described in #505

So, in short: your use-case will be covered as soon as #866 and #505 are fixed.

**Comment by pkozlowski-opensource** [comment]
OK, so with b39d2c0101491006979c7d7b8436edd9c25d32bf it should be possible to bind to both [className](https://developer.mozilla.org/en-US/docs/Web/API/Element/className) (`<div class-name="exp">`. I don't think that binding to [classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) will work as `classList` is a mutable structure.

The problem with binding to `className`, though, is that it will override any class names set using the `class` attribute - something to be aware / careful about.

@mhevery shell we come back to the idea of having more-like `ng-class` behaviour where people could write: `<div [class]="exp">`, where the `exp` could evaluate to an object (not sure if supporting strings and arrays as in 1.x is what we want).

**Comment by mhevery** [comment]
@pkozlowski-opensource yes, we should be bale to 

```
<div [class]=" {'my-class': true} ">
```

should work as `ng-class`

Can you make that happen.

**Comment by matanlurey** [comment]
Any update? This should also support `style` (per misko)

**Comment by pkozlowski-opensource** [comment]
This one is next on my list.
